cloudpickle/cloudpickle.py

```diff
@@ -434,7 +434,10 @@ def _save_subimports(self, code, top_level_dependencies):
             if isinstance(x, types.ModuleType) and hasattr(x, '__package__') and x.__package__:
                 # check if the package has any currently loaded sub-imports
                 prefix = x.__name__ + '.'
-                for name, module in sys.modules.items():
+                # A concurrent thread could mutate sys.modules,
+                # make sure we iterate over a copy to avoid exceptions
+                modules = sys.modules.copy()
```

Comment:
Is this actually thread-safe, per https://bugs.python.org/msg302327? I know `list(sys.modules.items())` is *not* (my testing showed this...)

Reply:
i.e., might we have to do this?
    
    names = list(sys.modules)
    for name in keys:
        module = sys.modules[name]

Reply:
I believe `.copy()` must acquire the GIL and therefore makes this thread-safe as a consequence but I am not 100% sure.

Reply:
Hmm, indeed, we'll need to use `list(sys.modules)` here.